### ISSUE TITLE
Excessive number of DNS requests

### ISSUE THREAD

**Issue by sharkannon** [Brandawg93/homebridge-nest-cam#66]
I have homebridge running through my personal DNS server and have noticed a major uptick in DNS requests since I setup homebridge-nest-cam.  I am seeing ~100k requests/day going to (each).

nexusapi-us1.camera.home.nest.com
webapi.camera.home.nest.com

That by farm exceeds any other requests I have at home by a factor of 10.  Though it's not really hurting anything, I know for a lot of users depending on their ISP, DNS requests like that could cause them to get blocked for short times as it could be considered an attack (Plus it's just pure inefficient)

I current run 4 Cams (1 IQ Outdoor, 1 Outdoor, 1 Indoor and a  Hello Cam)

Include with your bug report this version info:
```
node --version
v12.16.3
homebridge --version
1.1.0
homebridge-nest-cam v2.3.3

docker --version
Docker version 19.03.9, build 9d988398e7
```

**Comment by Brandawg93** [comment]
The more cameras you have, the more requests are needed to check for alerts and update camera info. You can turn these off in the advanced settings and it will not make these requests.

**Comment by sharkannon** [comment]
That might be true, but the TTL on these domains are 90s.  If you implemented dns caching on your app you would reduce the calls in my scenario from 100,000 to ~1000.  Nodejs isn't really my thing, I'm more of a Python person, but I know there are dns caching usually built in or available to most libs.

**Comment by Brandawg93** [comment]
Wouldn’t something like [unbound](https://nlnetlabs.nl/projects/unbound/about/) solve this issue? I run unbound via a docker container and my requests are very low. DNS cache seems somewhat beyond the scope of this plugin. I’ll take a look and see if there are any simple flags I could use on the request service I’m using, but something like unbound would solve this at the router level vs application level.

**Comment by sharkannon** [comment]
I have my own DNS server which basically does the same thing, not really the point though.  An application like this should be caching and using the TTL's of the dns and not resolving them on every  request (Kinda the point of TTL's).  It adds unnecessary load on the infrastructure by sending out dns requests for the same domain name every time.

**Comment by Brandawg93** [comment]
I wasn't able to find anything regarding caching DNS requests, but I did manage to cut the number of requests sent significantly. The plugin now updates all cameras at once and only updates camera info every minute (alerts are still per camera and every 10 seconds). I would appreciate it if you could test this for me with a test version before I publish this for everyone. You can install it via `npm -g i homebridge-nest-cam@test`. Let me know if you see the requests drop and thanks again for the heads up!

**Comment by sharkannon** [comment]
What about using something like https://www.npmjs.com/package/dnscache?  with https://github.com/request/request/issues/2536?

**Comment by sharkannon** [comment]
I have installed your test version however, we'll see how this works.  I'll let you know

**Comment by Brandawg93** [comment]
The plugin uses [axios](https://www.npmjs.com/package/axios) which would need to implement dnscache. I found a few possible plugins for axios, but I’d like to resolve the issue by simply cutting the number of requests. Thanks for trying it out.

**Comment by sharkannon** [comment]
ahh then something like https://www.npmjs.com/package/axios-cached-dns-resolve.

**Comment by Brandawg93** [comment]
Ya. That’s the one I found. I’m not sure if I want to use something with such a small user base, but I’ll play around with it. After some quick math, you should see your requests drop by almost half.

**Comment by sharkannon** [comment]
Looking since I implemented your update, DNS queries have dropped maybe 10% (with 200k requests that's not a lot) :(

**Comment by Brandawg93** [comment]
Have you tried disabling the advanced settings to see if the queries drop? I've discussed this with other developers, and their plugins have the same issue. At this time I'd prefer not to add another dependency that could potentially break others' setups over DNS resolutions.

**Comment by sharkannon** [comment]
Yes and to no real effect.  I use the motion alert but not the rest and disablign them all but the one didn't really seem to make much of a difference.. I'm assuming if any are enabled you query all anyway (guessing they're all in a single json blob anyway)

**Comment by Brandawg93** [comment]
Correct. If any are enabled it will poll every 10 seconds because the call for each of them is the same. I would love to be able to use push notifications like the Nest app, but it just isn't available for homebridge at the moment.

**Comment by Brandawg93** [comment]
I've published a new test version that allows you to configure how often the plugin checks for alerts. You can install it via the same command above. Let me know if that helps any.

**Comment by Brandawg93** [comment]
The test version is now published as v2.4.1.
